src/dataviews/helpers/histogram-helper.js

```diff
@@ -40,24 +75,53 @@ helper.fillNumericBuckets = function (buckets, start, width, numberOfBins) {
   }
 };
 
-function subtractAggregation (timestamp, aggregation, units) {
-  var date = moment.unix(timestamp).utc();
-  return date.subtract(units, MOMENT_AGGREGATIONS[aggregation]).unix();
-}
+helper.hasChangedSomeOf = function (list, changed) {
+  return _.some(_.keys(changed), function (key) {
+    return _.contains(list, key);
+  });
+};
 
-helper.calculateStart = function (buckets, start, aggregation) {
-  if (buckets.length === 0) {
-    return start;
-  }
+helper.calculateLimits = function (bins) {
+  var start = Infinity;
+  var end = -Infinity;
+
+  _.each(bins, function (bin) {
+    if (_.isFinite(bin.min)) {
+      start = Math.min(bin.min, start);
+    }
+    if (_.isFinite(bin.max)) {
+      end = Math.max(bin.max, end);
+    }
+  });
 
-  var firstBucket = buckets[0];
-  return subtractAggregation(start, aggregation, firstBucket.bin);
+  return {
+    start: start !== Infinity ? start : null,
+    end: end !== Infinity ? end : null
```

Comment:
Good catch. It should be compared to  `-Infinity`.

The use of those values is to pass it through `Math.max` and `Math.bin` and getting the first integer value and the first min or max. Any number will be greater than `-Infinity` and lesser than `Infinity`.